tests/lldp/test_lldp_syncd.py

```diff
@@ -346,6 +341,12 @@ def test_lldp_entry_table_after_reboot(
         safe_reboot=True,
         check_intf_up_ports=True
     )
+
+    # Wait till we have all lldp table entries in the DB. It's found in scaling
+    # setup this may take some time to be in sync.
+    result = wait_until(60, 5, 0, check_lldp_table_keys,
+                        db_instance, len(lldp_entry_keys_before_reboot))
+    pytest_assert(result, "Cannot get full lldp table")
```

Comment:
Based on my understanding of the previous review that tried to fix this https://github.com/sonic-net/sonic-mgmt/pull/15331 this won't always solve the problem.

The previous issue was that `eth0` can sometimes be slow to show up in the `LLDP_ENTRY_TABLE` after reboot (even if the previous test was a reboot test).
So capturing the `LLDP_ENTRY_TABLE` keys at the start of the test could be missing `eth0` so you'd have -1 entry in your `lldp_entry_keys_before_reboot`. Then after reboot if `eth0` shows up fast enough the `wait_until` might never return True.

Instead I'd propose that we leave `check_lldp_table_keys` the same and move the existing `wait_until` to after the reboot.

I.E.
```
    reboot(
        duthost,
        localhost,
        reboot_type=REBOOT_TYPE_COLD,
        reboot_helper=None,
        reboot_kwargs=None,
        safe_reboot=True,
        check_intf_up_ports=True
    )

+   keys_match = wait_until(30, 5, 0, check_lldp_table_keys, duthost, db_instance)
+   if not keys_match:
+       assert keys_match, "LLDP_ENTRY_TABLE keys do not match 'show lldp table' output"
```

Reply:
Code changed based on comment. Move wait_until after reboot and set delay of 30 seconds before check.